### PR TITLE
Select-into identity issue fixed for bigint, numeric, decimal

### PR DESCRIPTION
**Pull request by deepakshi-mittal**
Added different functions to handle int, smallint and bigint input 
Added test cases covering join, all numeral-data types and temp tables 
Changed error message for unsupported datatypes(i.e. non-numeral types) 
Numeric and decimal to be treated as bigint, same as that of create table behavior with identity column

Task: BABEL-4315

### Issues Resolved
Select-into identity issue fixed for bigint, numeric, decimal

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c [hunk 1]
```diff
@@ -236,29 +236,32 @@ TsqlFunctionIdentityInto(TypeName *typename, Node *seed, Node *increment, int lo
...
+			break;
+		case INT8OID:
+		case NUMERICOID:
+			args = list_make3((Node *)makeIntConst((int)INT8OID, location), seed, increment); /* Used bigint internally for decimal and numeric as well*/
+			result = (Node *)makeFuncCall(TsqlSystemFuncName("identity_into_bigint"), args, COERCE_EXPLICIT_CALL, location);
+			break;
+		default:
+			ereport(ERROR,
 					(errcode(ERRCODE_SYNTAX_ERROR),
-					 errmsg("identity column type must be smallint, integer, bigint, or numeric")));
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c:255]: Did you add a test case for this error ?

File: contrib/babelfishpg_tsql/src/pl_handler.c [hunk 2]
```diff
@@ -5094,72 +5130,65 @@ static List *transformSelectIntoStmt(CreateTableAsStmt *stmt, const char *queryS
...
 				arg_num = 0;
 				foreach (arg, funcexpr->args)
 				{
-					Node *fargNode = (Node *)lfirst(arg);
-					Const *con;
+					Node *farg_node = (Node *)lfirst(arg);
 					arg_num++;
 					switch (arg_num)
 					{
-						case 1:
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/pl_handler.c:5136]: We should throw some error when arg_num reach 4 or more in here ( although it's not possible , but for safe purpose)

File: contrib/babelfishpg_tsql/src/pl_handler.c [hunk 3]
```diff
@@ -5064,12 +5065,47 @@ static char *get_oid_type_string(int type_oid){
...
+			fxpr = (FuncExpr *)node;
+			if ((fxpr->args)->length != 1)
+				elog(ERROR, "Incorrect syntax near 'identity'");
+			n = (Node *)list_nth(fxpr->args, 0);
+			val = get_identity_into_args(n);
+			break;
+		case T_OpExpr:
+			opxpr = (OpExpr *)node;
+			if ((opxpr->args)->length != 1)
+				elog(ERROR, "Incorrect syntax near 'identity'");
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/pl_handler.c]: We should have a test regarding this code block

File: contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c (hunk 1 above)
Comment (deepakshi-mittal) [reply contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c:255]: Yes, I added this one:

SELECT col1, IDENTITY(char, 1,1) AS id_num INTO babel_539NewTable1 FROM babel_539OldTable;
GO
~~ERROR (Code: 33557097)~~
~~ERROR (Message: identity column type must be of data type int, bigint, smallint, or decimal or numeric)~~

**Review by forestkeeper — DISMISSED** [review]
LGTM, make sure the non-error behavior is the same with Sql server, rest are OK

**Review by forestkeeper — APPROVED** [review]

**Pull request merged**